unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuilderProjectsPanel/Scripts/Unpublish/UnpublishPopupController.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections;
+using DCL;
+using DCL.Interface;
+using UnityEngine;
+
+internal class UnpublishPopupController : IDisposable
+{
+    private const string TITLE = "Unpublish Scene";
+    private const string DESCRIPTION = "Are you sure you want to unpublish this scene?";
+    private const string SUCCESS_DESCRIPTION = "Scene unpublished";
+    private const string ERROR_TITLE = "Error";
+    private const string PROGRESS_TITLE = "Unpublishing...";
+
+    private IUnpublishPopupView view;
+    private Vector2Int coordinates;
+    private Coroutine fakeProgressRoutine = null;
+
+    public UnpublishPopupController(IUnpublishPopupView view)
+    {
+        this.view = view;
+        view.OnCancelPressed += OnCancel;
+        view.OnConfirmPressed += OnConfirmUnpublish;
+        view.Hide();
+    }
+
+    public void Dispose()
+    {
+        DataStore.i.builderInWorld.unpublishSceneResult.OnChange -= OnSceneUnpublished;
+        CoroutineStarter.Stop(fakeProgressRoutine);
+        view.OnCancelPressed -= OnCancel;
+        view.OnConfirmPressed -= OnConfirmUnpublish;
+        view.Dispose();
+    }
+
+    public void Show(Vector2Int coordinates)
+    {
+        this.coordinates = coordinates;
+        view.Show(TITLE, DESCRIPTION);
+    }
+
+    void OnConfirmUnpublish()
+    {
+        DataStore.i.builderInWorld.unpublishSceneResult.OnChange += OnSceneUnpublished;
+        WebInterface.UnpublishScene(coordinates);
+        fakeProgressRoutine = CoroutineStarter.Start(ProgressRoutine());
+    }
+
+    void OnCancel()
+    {
+        view.Hide();
+        CoroutineStarter.Stop(fakeProgressRoutine);
+    }
+
+    IEnumerator ProgressRoutine()
+    {
+        float progress = 0;
+        while (true)
+        {
+            progress = Mathf.Clamp(progress + UnityEngine.Random.Range(0.01f, 0.03f), progress, 0.99f);
```

Comment:
```suggestion
            progress = Mathf.Clamp(progress + UnityEngine.Random.Range(0.01f, 0.03f), 0, 0.99f);
```
Shouldnt it be like this?

Reply:
It's kind of the same thing case `progress` should never decrease... but I'll change that, I agree is more confusing this way 🤣